seafile-seahub: Broken Dependencies - flup

2019-06-26

* git pull [master]
* scripts/feeds update -a
* scripts/feeds install -a

diffconfig, after pusleaudio errors, returns
```
CONFIG_TARGET_ar71xx=y
CONFIG_TARGET_ar71xx_generic=y
CONFIG_TARGET_ar71xx_generic_DEVICE_mr16=y
```
make indicates 
```
WARNING: Makefile 'package/feeds/packages/seafile-seahub/Makefile' has a dependency on 'flup', which does not exist
 make[1] clean
WARNING: Makefile 'package/feeds/packages/seafile-seahub/Makefile' has a dependency on 'flup', which does not exist
make[2]: Entering directory '/home/jeff/devel/openwrt/scripts/config'
make[2]: Leaving directory '/home/jeff/devel/openwrt/scripts/config'
```

Reported by OP at https://forum.openwrt.org/t/meraki-mr16-noob-to-unix-please-help/39172/22?u=jeff and locally confirmed with my existing build tree.

```
jeff@deb-devel:~/devel/openwrt$ make menuconfig
Collecting package info: done
Collecting target info: done
WARNING: Makefile 'package/feeds/packages/seafile-seahub/Makefile' has a dependency on 'flup', which does not exist
tmp/.config-package.in:102870:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:102870:	symbol PACKAGE_baresip-mod-pulse depends on PACKAGE_baresip-mod-pulse
tmp/.config-package.in:121610:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:121610:	symbol PACKAGE_pulseaudio-profiles depends on PACKAGE_pulseaudio-profiles
tmp/.config-package.in:121625:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:121625:	symbol PACKAGE_pulseaudio-tools depends on PACKAGE_pulseaudio-tools


*** End of the configuration.
*** Execute 'make' to start the build or try 'make help'.

jeff@deb-devel:~/devel/openwrt$ ./scripts/diffconfig.sh 
tmp/.config-package.in:102870:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:102870:	symbol PACKAGE_baresip-mod-pulse depends on PACKAGE_baresip-mod-pulse
tmp/.config-package.in:121610:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:121610:	symbol PACKAGE_pulseaudio-profiles depends on PACKAGE_pulseaudio-profiles
tmp/.config-package.in:121625:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:121625:	symbol PACKAGE_pulseaudio-tools depends on PACKAGE_pulseaudio-tools
tmp/.config-package.in:102870:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:102870:	symbol PACKAGE_baresip-mod-pulse depends on PACKAGE_baresip-mod-pulse
tmp/.config-package.in:121610:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:121610:	symbol PACKAGE_pulseaudio-profiles depends on PACKAGE_pulseaudio-profiles
tmp/.config-package.in:121625:error: recursive dependency detected!
For a resolution refer to Documentation/kbuild/kconfig-language.txt
subsection "Kconfig recursive dependency limitations"
tmp/.config-package.in:121625:	symbol PACKAGE_pulseaudio-tools depends on PACKAGE_pulseaudio-tools
CONFIG_TARGET_ar71xx=y
CONFIG_TARGET_ar71xx_generic=y
CONFIG_TARGET_ar71xx_generic_DEVICE_mr16=y
jeff@deb-devel:~/devel/openwrt$ make -j12 clean download world
WARNING: Makefile 'package/feeds/packages/seafile-seahub/Makefile' has a dependency on 'flup', which does not exist
 make[1] clean
^C/home/jeff/devel/openwrt/include/toplevel.mk:193: recipe for target 'clean' failed
make: *** [clean] Interrupt
```

--- a/net/seafile-seahub/Makefile
+++ b/net/seafile-seahub/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=seafile-seahub
 PKG_VERSION:=6.3.4
-PKG_RELEASE:=4
+PKG_RELEASE:=5
 PKG_LICENSE:=Apache-2.0
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
@@ -35,7 +35,7 @@ define Package/seafile-seahub
 	+django-compressor +django-formtools +django-jsonfield \
 	+django-picklefield +django-postoffice +django-restframework \
 	+django-simple-captcha +django-statici18n +django-webpack-loader \
-	+flup +gunicorn +openpyxl \
+	+python-flup +gunicorn +openpyxl \
 	$(foreach dep,$(SEAFILE_PYTHON_DEPENDS),+python-$(dep))
 endef
 

